lib/datadog/error_tracking/component.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+require 'set'
+require_relative 'collector'
+require_relative 'filters'
+
+module Datadog
+  module ErrorTracking
+    # Component for Error Tracking.
+    #
+    # Only one instance of the Component should ever be active.
+    #
+    # The component instance records every handled exceptions from the configured scopes
+    # (user, third_party packages, specified files or everything).
+    class Component
+      LOCK = Mutex.new
+
+      attr_accessor :handled_exc_tracker,
+        :module_path_getter,
+        :tracer,
+        :handled_errors_include,
+        :instrumented_files
+
+      class << self
+        def build(settings, tracer, logger)
+          return if !settings.respond_to?(:error_tracking) || (settings.error_tracking.handled_errors.nil? &&
+            settings.error_tracking.handled_errors_include.empty?)
+
+          return unless environment_supported?(logger)
+
+          new(
+            tracer: tracer,
+            handled_errors: settings.error_tracking.handled_errors,
+            handled_errors_include: settings.error_tracking.handled_errors_include,
+          ).tap(&:start)
+        end
+
+        # Checks whether the runtime environment is supported by
+        # error tracking.
+        def environment_supported?(logger)
+          if RUBY_ENGINE != 'ruby'
+            logger.warn("error tracking: cannot enable error tracking: MRI is required, but running on #{RUBY_ENGINE}")
+            return false
+          end
+          if RUBY_VERSION < '2.6'
+            logger.warn(
+              "error tracking: cannot enable error tracking: Ruby 2.6+ is required, but running
+              on #{RUBY_VERSION}"
+            )
+            return false
+          end
+          true
```

Comment:
I think there is not benefit to go guard-clause for such composition

```suggestion
          if RUBY_ENGINE != 'ruby'
            logger.warn("error tracking: cannot enable error tracking: MRI is required, but running on #{RUBY_ENGINE}")
            false
          elsif RUBY_VERSION < '2.6'
            logger.warn(
              "error tracking: cannot enable error tracking: Ruby 2.6+ is required, but running
              on #{RUBY_VERSION}"
           )
           false
         else
           true
         end
```